tests/components/analytics/doughnut_chart.test.jsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+
+import {shallow, mount} from 'enzyme';
+
+import DoughnutChart from 'components/analytics/doughnut_chart.jsx';
+
+jest.mock('chart.js');
+
+describe('components/analytics/doughnut_chart.jsx', () => {
+    test('should match snapshot, on loading', () => {
+        const wrapper = shallow(
+            <DoughnutChart
+                title='Test'
+                height={400}
+                width={600}
+            />
+        );
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should match snapshot, loaded without data', () => {
+        const data = {};
+
+        const wrapper = shallow(
+            <DoughnutChart
+                title='Test'
+                height={400}
+                width={600}
+                data={data}
+            />
+        );
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    test('should match snapshot, loaded with data', () => {
+        const data = {
+            datasets: [
+                {data: [1, 2, 3]}
+            ]
+        };
+
+        const wrapper = shallow(
```

Comment:
Should probably use `mountWithIntl` instead of `shallow` as I couldn't see the difference in snapshots between "loaded without data" and "loaded with data".

Reply:
This won't work, becase the canvas content isn't showed anyway, probably the best aproach to test it is to mock the Chart object and assert that is called with the received data.

Reply:
actually, I have already mocked the Chart library because the test doesn't support 2d drawing contexts, so, we have to mock anyway.

Reply:
Fixed! 😃

Reply:
Interestingly, below is what I've got using `mountWithIntl`. Same with what you've got, just minus the `intl`. Anyway, we'll go with your approach.

```
exports[`components/analytics/doughnut_chart.jsx should match snapshot, loaded with data 1`] = `
<DoughnutChart
  data={
    Object {
      "datasets": Array [
        Object {
          "data": Array [
            1,
            2,
            3,
          ],
        },
      ],
    }
  }
  height={400}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  title="Test"
  width={600}
>
  <div
    className="col-sm-6"
  >
    <div
      className="total-count"
    >
      <div
        className="title"
      >
        Test
      </div>
      <div
        className="content"
      >
        <canvas
          height={400}
          width={600}
        />
      </div>
    </div>
  </div>
</DoughnutChart>
`;

exports[`components/analytics/doughnut_chart.jsx should match snapshot, loaded without data 1`] = `
<DoughnutChart
  data={Object {}}
  height={400}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  title="Test"
  width={600}
>
  <div
    className="col-sm-6"
  >
    <div
      className="total-count"
    >
      <div
        className="title"
      >
        Test
      </div>
      <div
        className="content"
      >
        <canvas
          height={400}
          width={600}
        />
      </div>
    </div>
  </div>
</DoughnutChart>
`;
```

Reply:
Yes, the reason I used mount and not mountWithIntl, is because for the cases that I was testing wasn't needed the i18n, so isn't needed the intl prop.